hey.go

```diff
@@ -190,7 +196,36 @@ func main() {
 			usageAndExit(err.Error())
 		}
 	}
+	// at this point we will read through the CSV and then use that as the url
+
+	csvfile, err := os.Open(fileName)
+	if err != nil {
+		log.Fatalln("couldnt open the csv file", err)
+	}
+	r := csv.NewReader(csvfile)
+	// TODO remove this after i'm done testing
+	break_early := true
+	for {
+		record, err :=  r.Read()
+		nurl := record[0]
+		// ignore any strings that aren't formatted correctly
+		if strings.Contains(nurl, ".com") == false {
+			continue;
+		}
+		fmt.Printf("the url is: %s\n", nurl)
+
+		//err == io.EOF or
+		if err != nil {
+			log.Fatal(err)
+		}
+		generateRequestAndGetResponse(nurl, method, username, password, *hostHeader, header, dur, bodyAll, num, conc, rateLimit, proxyURL)
+		if break_early {
+			break
+		}
+	}
+}
 
+func generateRequestAndGetResponse(url string, method string, username string, password string, hostHeader string, header http.Header, dur time.Duration, bodyAll []byte, num int, conc int, rateLimit float64, proxyURL *gourl.URL) {
```

Comment:
This seems like it would repeat w.Init/w.Run for each CSV line: the approach should probably keep the initialization and start of workers as the main single execution (e.g. like worker.py in GKEmail).
I've put together an alternative hack in https://github.com/rakyll/hey/pull/245 that spins off a reader goroutine that feeds the workers urls through a shared channel.